librz/il/il_vm_eval.c

```diff
@@ -291,20 +291,34 @@ RZ_API void rz_il_vm_clear_events(RzILVM *vm) {
  * \param vm pointer to VM
  * \param op_list, a list of op roots.
  * \param fallthrough_addr initial address to set PC to. Thus also the address to "step to" if no explicit jump occurs.
+ * \param delayed_by it is used to notify the vm_step to update the pc after N instructions.
  */
-RZ_API bool rz_il_vm_step(RzILVM *vm, RzILOpEffect *op, ut64 fallthrough_addr) {
+RZ_API bool rz_il_vm_step(RzILVM *vm, RzILOpEffect *op, ut64 fallthrough_addr, ut32 delayed_by) {
 	rz_return_val_if_fail(vm && op, false);
 
 	rz_il_vm_clear_events(vm);
 
 	// Set the successor pc **before** evaluating. Any jmp/goto may then overwrite it again.
-	RzBitVector *next_pc = rz_bv_new_from_ut64(vm->pc->len, fallthrough_addr);
+	RzBitVector *next_pc = NULL;
```

Comment:
Not very happy about this solution, to be honest

Reply:
i'm exploring possibilities.